src/symbolize/gimli/parse_running_mmaps_unix.rs

```diff
@@ -76,6 +76,11 @@ impl MapsEntry {
     pub(super) fn ip_matches(&self, ip: usize) -> bool {
         self.address.0 <= ip && ip < self.address.1
     }
+
+    #[cfg(target_os = "android")]
+    pub(super) fn offset(&self) -> usize {
+        self.offset
+    }
```

Comment:
Good catch, thanks! I didn't think about it and just exposed a getter for the existing field, but it should probably use a `u64` instead, for exactly the scenario you described.

> can this entire PR only affect 64-bit Android?

Not without "artificially" leaving 32-bit Android unsupported; even the latest version of Android still supports 32-bit-only devices, and the embedded-library mechanism has been present in both 32-bit and 64-bit Android devices since Android 6 (API level 23), ~9 years ago.

Reply:
hopefully no one is creating 5GB APKs but, well, I've met programmers before

Reply:
(IIRC I read somewhere that APKs over 2GB are disallowed, but that's not really something we should rely on anyways)